workers/unity/Packages/io.improbable.gdk.core/Subscriptions/CommandSenderReceiverSubscriptionManagerBase.cs

```diff
@@ -124,19 +124,22 @@ public override void ResetValue(ISubscription subscription)
     public abstract class CommandReceiverSubscriptionManagerBase<T> : SubscriptionManager<T>
         where T : ICommandReceiver
     {
-        private readonly ComponentUpdateSystem componentUpdateSystem;
+        private readonly EntityManager entityManager;
 
         private Dictionary<EntityId, HashSet<Subscription<T>>> entityIdToReceiveSubscriptions;
 
         private HashSet<EntityId> entitiesMatchingRequirements = new HashSet<EntityId>();
         private HashSet<EntityId> entitiesNotMatchingRequirements = new HashSet<EntityId>();
 
-        private readonly uint componentId;
+        private readonly ComponentType componentType;
+        private readonly ComponentType componentAuthType;
 
         protected CommandReceiverSubscriptionManagerBase(World world, uint componentId) : base(world)
         {
-            this.componentId = componentId;
-            componentUpdateSystem = world.GetExistingSystem<ComponentUpdateSystem>();
+            componentType = ComponentDatabase.Metaclasses[componentId].Data;
+            componentAuthType = ComponentDatabase.Metaclasses[componentId].Authority;
```

Comment:
Use `ComponentDatabase.GetMetaclass(componentId)` instead? It gives a more helpful error if this ever was to throw an error